DEHPCommon/HubController/HubController.cs

```diff
@@ -392,39 +392,92 @@ public IEnumerable<NestedElement> GetNestedElementTree(Option option, DomainOfEx
         /// <param name="iteration"></param>
         /// <param name="domain"></param>
         public async Task Upload(File file = null, Iteration iteration = null, DomainOfExpertise domain = null)
+        {
+            await Upload(null, file, iteration, domain);
+        }
+
+        /// <summary>
+        /// Upload one file to the <see cref="DomainFileStore"/> of the specified domain or of the active domain
+        /// </summary>
+        /// <param name="filePath"></param>
+        /// <param name="file"></param>
+        /// <param name="iteration"></param>
+        /// <param name="domain"></param>
+        public async Task Upload(string filePath, File file = null, Iteration iteration = null, DomainOfExpertise domain = null)
         {
             iteration ??= this.GetIteration().Keys.First();
             domain ??= this.Session.QueryCurrentDomainOfExpertise();
-            var iDalUri = new Uri(this.Session.DataSourceUri);
 
             var fileStore = iteration.DomainFileStore.FirstOrDefault(x => x.Owner.Iid == domain.Iid);
 
-            if (fileStore is null || !this.GetFile(out var filePath, out var fileName, out var extensions))
+            if (fileStore is null)
             {
+                this.logger.Error("No DomainFileStore found");
                 return;
             }
             
+            var iDalUri = new Uri(this.Session.DataSourceUri);
+            var currentParticipant = this.OpenIteration.GetContainerOfType<EngineeringModel>().GetActiveParticipant(this.Session.ActivePerson);
+
+            string fileName;
+            string[] extensions;
+
+            if (String.IsNullOrEmpty(filePath))
```

Comment:
It's better to use string.IsNullOrWhiteSpace, It checks for null, empty, and whitespace

Reply:
I was not aware of a difference in the implementation between `string.IsNullOrEmpty()` and `String.IsNullOrEmpty() `, but in fact I wanted to use `string`.